src/communication/stream_transfer_sender.hpp

```diff
@@ -72,6 +78,8 @@ namespace cubstream
       cubthread::daemon *m_sender_daemon;
       char m_buffer[cubcomm::MTU];
 
+      bool m_is_termination_phase;
```

Comment:
I think `m_is_termination_phase` is set on true in `enter_termination_phase` function from different thread, wouldn't be better to change the type to `std::atomic<bool>`